### PR TITLE
ThreatConnect v2 - Support multiple owners in fetching indicators

### PR DESCRIPTION
**Pull request by adi88d**
## XSOAR [Partner](https://xsoar.pan.dev/docs/partners/become-a-tech-partner)?
- [ ] The title must be in the following format: **[YOUR_PARTNER_ID] short description**.

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes:
- https://github.com/demisto/etc/issues/27910

## Description
1.  When fetching indicators from multiple owners, only one result from one owner returns, updated get_indicators and create_context methods fixed this.
* Note that when there is an indicator with multiple dbot score(from multiple owners), the indicator score in demisto is the highest one.

2. Updated ***confidence*** parameter default value to 3

## Minimum version of Demisto
- [x] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Tests/conf.json [hunk 1]
```diff
@@ -3152,7 +3151,6 @@
         "_comment2": "~~~ UNSTABLE ~~~",
         "ServiceNow": "Instance goes to hibernate every few hours",
         "Tenable.sc": "unstable instance",
-        "ThreatConnect v2": "unstable instance",
```
Comment (esharf) [inline Tests/conf.json:3155]: why are you removing from skipped?

**Review by esharf — APPROVED** [review]

File: Tests/conf.json (hunk 1 above)
Comment (adi88d) [reply Tests/conf.json:3155]: To be sure the test playbook passed in the build process too

Comment (esharf) [reply Tests/conf.json:3155]: so don’t forget to skip the test after you merge this PR.

File: Packs/ThreatConnect/Integrations/ThreatConnect_v2/ThreatConnect_v2.py [hunk 2]
```diff
@@ -114,13 +118,15 @@ def create_context(indicators, include_dbot_score=False):
         else:
             dbot_score = 1
 
-        if include_dbot_score:
-            context['DBotScore'].append({
+        # if there is more than one indicator results - take the one with the highest score
+        if include_dbot_score and dbot_score > highest_score:
```
Comment (yaakovi) [inline Packs/ThreatConnect/Integrations/ThreatConnect_v2/ThreatConnect_v2.py:122]: indicators list is not always the same indicator by different owners, it may be different indicators

File: Packs/ThreatConnect/Integrations/ThreatConnect_v2/ThreatConnect_v2.yml [hunk 3]
```diff
@@ -33,7 +33,7 @@ configuration:
   required: false
 - display: Confidence threshold for Malicious Indicators
   name: confidence
-  defaultvalue: "50"
+  defaultvalue: "3"
```
Comment (yaakovi) [inline Packs/ThreatConnect/Integrations/ThreatConnect_v2/ThreatConnect_v2.yml:36]: revert this change

File: Packs/ThreatConnect/ReleaseNotes/2_0_6.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### ThreatConnect v2
+- Fixed an issue where fetching indicators from multiple owners did not work as intended.
+- Updated ***confidence*** parameter default value to 3.
```
Comment (yaakovi) [inline Packs/ThreatConnect/ReleaseNotes/2_0_6.md:5]: remove

File: Packs/ThreatConnect/ReleaseNotes/2_0_6.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### ThreatConnect v2
+- Fixed an issue where fetching indicators from multiple owners did not work as intended.
```
Comment (yaakovi) [inline Packs/ThreatConnect/ReleaseNotes/2_0_6.md:4]: `fetching indicators from multiple owners is now supported.`

@kirbles19 what do you think?

**Review by yaakovi — APPROVED** [review]

File: Packs/ThreatConnect/ReleaseNotes/2_0_6.md (hunk 5 above)
Comment (kirbles19) [reply Packs/ThreatConnect/ReleaseNotes/2_0_6.md:4]: Added support for fetching indicators from multiple owners.

**Comment by chris4490** [comment]
@adi88d same customer reported the issue. They're using the IP command and it's not returning data from multiple owners:

Looking at the latest code, the break is still within the get_indicators code. Should this have been removed to resolve this?

Threatconnect1

Threatconnect

image
